RUFAS/biophysical/manure/storage/solids_storage_calculator.py

```diff
@@ -0,0 +1,213 @@
+import math
+from RUFAS.biophysical.manure.storage.composting_type import CompostingType
+
+
+FRACTION_NITROGEN_LOST_TO_LEACHING: dict[CompostingType, float] = {
+    CompostingType.STATIC_PILE: 0.06,
+    CompostingType.PASSIVE_WINDROW: 0.04,
+    CompostingType.INTENSIVE_WINDROW: 0.06,
+}
+
+DEFAULT_CARBON_FRACTION_AVAILABLE_IN_VSD: float = 0.5
+"""Default carbon content (percent by mass) of manure degradable volatile solids (unitless, [0, 1])."""
+
+DEFAULT_CARBON_FRACTION_AVAILABLE_IN_VSND: float = 0.35
+"""Default carbon content (percent by mass) of manure non-degradable volatile solids (unitless, [0, 1])."""
+
+DEFAULT_EFFECT_OF_MOISTURE_ON_MICROBIAL_DECOMPOSITION: float = 0.65
+"""The default effect of moisture on microbial decomposition."""
+
+EFFECTIVENESS_OF_MICROBIAL_DECOMPOSITION_RATE: float = 2.37e-3
+"""The rate of effectiveness of microbial decomposition."""
+
+FIRST_ORDER_DECAYING_COEFFICIENT: float = 0.1
+"""The first order decaying coefficient."""
+
+DEFAULT_LAG_TIME: int = 2
+"""Default lag time used in the calculation of the carbon decomposition rate (days). Default is set to 2."""
+
+DEFAULT_DAYS_SINCE_LAST_TURNING: int = 1
+"""Default days since the previous compost turning event (days). Default is set to 1."""
+
+COMPOSTING_DECOMPOSITION_TEMPERATURE: float = 60.0
+"""The temperature of the inner compost layer at which microbial growth and decomposition is maximized (C)."""
+
+DEFAULT_MOLE_FRACTION_OF_OXYGEN: float = 0.15
+"""The default mole fraction of oxygen in the air wihtin the compost layer."""
+
+OXYGEN_HALF_SATURATION_CONSTANT: float = 0.02
+"""The half saturation constant of Oxygen gas (O2)"""
+
+AMBIENT_AIR_MOLE_FRACTION_OF_OXYGEN: float = 0.21
+"""The mole fraction of oxygen in ambient air."""
+
+
+class SolidsStorageCalculator:
+    """
+    This class contains methods to calculate the carbon decomposition, methane emission,
+    nitrogen loss to leaching, and dry matter loss of the current day.
+    The calculations are based on the composting type and the manure temperature.
+    The methods are static and can be called without creating an instance of the class.
+    """
+
+    @staticmethod
+    def _calculate_nitrogen_loss_to_leaching(composting_type: CompostingType, received_manure_nitrogen: float) -> float:
+        """
+        This function calculates the amount of nitrogen leached out of the manure-bedding
+        pile of the current day.
+
+        Parameters
+        ----------
+        composting_type : CompostingType
+            The type of composting being used.
+        received_manure_nitrogen : float
+            The nitrogen content of the received manure, kg.
+
+        Returns
+        -------
+        float
+            The total nitrogen loss to leaching of the current day, kg.
+        """
+
+        return FRACTION_NITROGEN_LOST_TO_LEACHING[composting_type] * received_manure_nitrogen
```

Comment:
```suggestion
        return fraction_nitrogen_lost_to_leaching * received_manure_nitrogen
```
The fraction should be passed in because the open lot/ CBPB does not use the dictionary lookup.